Tranks implementation

NathanZiri opened the pull request:
Added 2 tracks to the game with AI following the track path

NathanZiri committed:
fully implemented tracks and extra models

WendyFu004 committed:
Update Canyon Race Track

Added more nodes to the track and the track became longer now.
Added some ring tracks and AI birds to the Canyon race scene.
Put the target mark at the end of the track.

NathanZiri committed:
added thrusters

NathanZiri committed:
Merge branch 'TranksImplementation' of https://github.com/afourneaux/rocket-racing-476 into TranksImplementation

Aether700 reviewed (changes requested):
The mountain track does not have a target at the end of it nor does it have a bird spot where the racers can get attacked by birds. 

For the cayon, the rings in the cayon do not fit the esthetic of the game, would it be possible to use arrows like in mario kart to guide the racers instead of the rings?

Also, in the cayon track nothing stops the player from just flying above the cayon, cut directly accross straight to the target. Can you please look into making sure that is not possible.

Finally, the branch is not up to date with main. Can you please update the branch to pull from main. To double check that everything works well make sure that the racers get destroyed when they hit the target at the end of the track.

You will also need to set up the game over screen once that is all done. if you need help with that part let me know

NathanZiri commented:
I will discuss with yuwen to address these issues and will pull update the branch from main to ensure it is more upto dare when being re pushed.

NathanZiri committed:
Merge branch 'main' into TranksImplementation

WendyFu004 committed:
Time score added

Now the final score would included the time score and Final score = accuracyScore - timeScore
It was sum before I changed to minus here since time score is the time spent so far. The accuracy score is very huge now, we can balance the weight or find a new time score rule.

Aether700 commented on Assets/Scripts/PositionTracker.cs:
```diff
@@ -69,6 +73,7 @@ private void FinishRace(Bullseye bullseyeObj, Vector3 bullseyeHitPoint)
         score.SetAccuracyScore(bullseyeObj.GetScore(bullseyeHitPoint));
 
         // Compute time score here
+        score.SetTimeScore((int)(Time.time - startTime));
```
Would it be possible to have the time score be a "positive" number? as in the greater the time score the greater the final score?

For example we could have the first racer to finish always having a set maximum score and then decrease the score of the other racers based on the time they take compared to that initial time?

WendyFu004 committed:
Update for two track scenes

-Added AI birds, bullseyes and game over UI for both Mountain and canyon scenes
-Added glass ceiling and wall to the canyon scenes  so player won't play through
TODO:
-Arrows to the track
-Mountain track need a mesh collider to restrict player's movement. Will talk to Nathan about this part.

Aether700 reviewed (changes requested):
In the mountain scene the bullseye is not at the end of the track. Please either update the track to end at the bullseye target or move the bullseye target at the end of the track. Also, most of the environment of the mountain scene has no colliders. Please add colliders to the scene and make sure that the AI avoids those colliders by using the correct layers of setting the layer mask of the AI correctly (I can help with that if necessary). Additionally the mountain track still has no birds on it. Please add some.

For the cayon scene, the bullseye is slightly off of the end of the track. you can easily modify that by having the bullseye target transform be the final position of the race track (I can help with that if you need to).

I also noticed that you are using mesh colliders for a lot of things when we can easily get away with other simpler colliders which run faster. For example. The plane collider in the mountain scene and the glass ceilling collider in the cayon scene both use mesh colliders which is really expensive when they could easily use box colliders. The same goes for the launch pad which would use a box collider.

WendyFu004 committed:
Correct the position of the bullseyes and more

Mountain race: I used mesh colliders for this scene since it is hard to find colliders fixes the irregular track shape.

Canyon race: change the mesh collider of glass ceiling and wall to box collider. Tweaked track nodes.

Both scene:
-Correct positions of bullseyes and the last node of track.
-Added box collider to the launchpads
-Added birds to the scene(only one bird AI zone so far)

Not don:
let AI avoids colliders by using the right layer. (I am not sure which layer we should use)

Aether700 reviewed (commented):
Good job. The Canyon in the canyon scene doesn't have any colliders on it. you can probably just do a mesh collider on that one given how complex it is.

For the mountain scene the bird area is kind of short. you should make it bigger. There is also a collider on a plane under the landing pad in the mountain scene which has a mesh collider on it. Given where in the scene it is you could probably just remove the collider but if you think it should keep it's collider use a box collider instead.

WendyFu004 committed:
update the bird area and plane collider

-Canyon scene added mesh collider on it
-Mountain scene created larger bird area and add more AI birds to it. Replaced the mesh collider on the gourd with the box one.

Aether700 commented:
There are merge conflicts preventing from properly pulling the changes. Can you please fix them?

WendyFu004 committed:
Merge branch 'main' into TranksImplementation

Aether700 reviewed (approved):
Great work
